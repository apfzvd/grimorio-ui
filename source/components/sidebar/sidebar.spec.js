import '../../../internals/test/helper';

import Sidebar from './sidebar-component';

const schema = [
  {
    name: 'Home',
    link: '/home',
    icon: 'person',
    id: 'home-link',
  },
  {
    name: 'PÃ¡gina 1',
    link: '/pag1',
    icon: 'desktop_mac',
    id: 'pag1',
  },
  {
    name: 'Um Accordion',
    icon: 'filter',
    id: 'acc',
    submenu: [
      {
        name: 'Item 1',
        link: '/acc/item1',
        id: 'acc-item1',
      },
     {
        name: 'Item 2',
        link: '/acc/item2',
        id: 'acc-item2',
      },
    ],
   },
   {
    name: 'Outro Accordion',
    icon: 'filter',
    id: 'acc2',
    submenu: [
      {
        name: 'Item 1',
        link: '/acc2/item1',
        id: 'acc2-item1',
      },
     {
        name: 'Item 2',
        link: '/acc2/item2',
        id: 'acc2-item2',
      },
    ],
   },
];

const initialProps = { schema, onClickItem: jest.fn(), initialSubmenu: 'acc2', initialItem: 'acc2-item2' };

describe('Sidebar component', () => {
  let wrapper

  beforeAll(() => {
    wrapper = shallow(
      <Sidebar {...initialProps} />
    );
  })

  it('render default', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  describe('with toggle button', () => {
    beforeAll(() => {
      wrapper.setProps({ hasToggle: true })
    })

    it('render with button when hasToggle is true', () => {
      expect(wrapper.find('.toggle')).toHaveLength(1);
    });

    it('should call onToggle when click on toggle', () => {
      const props = { onToggle: jest.fn() }
      wrapper.setProps(props)
      jest.spyOn(props, 'onToggle')
      wrapper.find('.toggle').simulate('click');
      expect(props.onToggle).toHaveBeenCalled();
    })
  })

  describe('with overlay', () => {
    beforeEach(() => {
      wrapper.setProps({ isMobile: true })
    })

    it('render with overlay when isMobile is true', () => {
      expect(wrapper.find('.overlay')).toHaveLength(1);
    });

    it('should call onClick when click on button', () => {
      const props = { onToggle: jest.fn() }
      wrapper.setProps(props)
      jest.spyOn(props, 'onToggle')
      wrapper.find('.overlay').simulate('click');
      expect(props.onToggle).toHaveBeenCalled();
    })
  })

  describe('with logo', () => {
    const logo = <span className="logo-default-teste">logo default</span>;
    const logoSmall = <span className="logo-small-teste">logoSmall</span>;

    beforeEach(() => {
      wrapper.setProps({ isMobile: false, logo, logoSmall})
    })

    it('render default logo when open is true', () => {
      wrapper.setProps({ open: true })
      expect(wrapper.find('.logo-default-teste')).toHaveLength(1);
    });

    it('render small logo when open is true', () => {
      wrapper.setProps({ open: false })
      expect(wrapper.find('.logo-small-teste')).toHaveLength(1);
    });

    it('should call onClick when click on button', () => {
      const props = { onLogoClick: jest.fn()}
      wrapper.setProps({ onLogoClick: props.onLogoClick })
      jest.spyOn(props, 'onLogoClick')
      wrapper.find('.logotype').simulate('click')
      expect(props.onLogoClick).toHaveBeenCalled();
    })
  })
});
