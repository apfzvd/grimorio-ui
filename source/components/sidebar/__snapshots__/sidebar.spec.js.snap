// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component render default 1`] = `
"<Symbol(react.fragment)>
  <div className=\\"sidebar closed\\">
    <div onClick={[undefined]} className=\\"logotype\\" />
    <nav className=\\"content\\">
      <span className=\\"contentTitle isNavClosed\\">
        Menu
      </span>
      <Accordion type=\\"accordionMenu\\" exclusive={false} as=\\"ul\\" open={[undefined]} theme=\\"default\\" panels={{...}}>
        <li title=\\"Home\\" className=\\"listItem isClosed\\" onClick={[Function]} data-testid=\\"home-link\\">
          <Icon size={16} name=\\"person\\" className=\\"iconLeft\\" align=\\"middle\\" />
          Home
        </li>
        <li title=\\"Página 1\\" className=\\"listItem isClosed\\" onClick={[Function]} data-testid=\\"pag1\\">
          <Icon size={16} name=\\"desktop_mac\\" className=\\"iconLeft\\" align=\\"middle\\" />
          Página 1
        </li>
        <li className=\\"accordeonListItem isClosed\\" data-testid=\\"acc\\">
          <AccordionTitle data-testid=\\"acc\\" active={false} index={2} onClick={[Function]} icon=\\"filter\\">
            Um Accordion
          </AccordionTitle>
          <AccordionContent active={false}>
            <ul className=\\"nestedMenu\\">
              <li title=\\"Item 1\\" className=\\"nestedItem isClosed\\" onClick={[Function]} data-testid=\\"acc-item1\\">
                Item 1
              </li>
              <li title=\\"Item 2\\" className=\\"nestedItem isClosed\\" onClick={[Function]} data-testid=\\"acc-item2\\">
                Item 2
              </li>
            </ul>
          </AccordionContent>
        </li>
        <li className=\\"accordeonListItem isActive isClosed\\" data-testid=\\"acc2\\">
          <AccordionTitle data-testid=\\"acc2\\" active={true} index={3} onClick={[Function]} icon=\\"filter\\">
            Outro Accordion
          </AccordionTitle>
          <AccordionContent active={true}>
            <ul className=\\"nestedMenu\\">
              <li title=\\"Item 1\\" className=\\"nestedItem isClosed sectionActive\\" onClick={[Function]} data-testid=\\"acc2-item1\\">
                Item 1
              </li>
              <li title=\\"Item 2\\" className=\\"nestedItem isClosed isActive sectionActive\\" onClick={[Function]} data-testid=\\"acc2-item2\\">
                Item 2
              </li>
            </ul>
          </AccordionContent>
        </li>
      </Accordion>
    </nav>
  </div>
</Symbol(react.fragment)>"
`;
